staging/src/k8s.io/apimachinery/pkg/util/manifest/manifest.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubernetes
+
+import (
+	"bytes"
+	"io"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/yaml"
+)
+
+// NewUnstructuredList decodes YAML or JSON data into a flattened list of Kubernetes resources,
+// it will attempt parsing all manner of manifests, such as YAML multi docs and any list types
+func NewUnstructuredList(data []byte) (*unstructured.UnstructuredList, error) {
```

Comment:
Shall we name this `(Parse|Unmarshal|Decode)UnstructuredList`? I guess `Parse...` would make the most sense.

Reply:
I guess it's better also to accept an `io.ReadCloser` instead of bytes here, to allow multiple kinds of sources?

Reply:
If taking a reader is important, I'd insist on two functions (one with `[]bytes` and another with a reader).